File: ssr-simple/src/main/java/com/cincinnatiai/ssr_simple/model/ModifierModel.kt
package com.cincinnatiai.ssr_simple.model

data class ModifierModel(
    val height: Int? = null,
    val width: Int? = null,
    val weight: Float? = null, // For Row/Column children: proportional size
    val padding: Int? = null,
    val paddingTop: Int? = null,
    val paddingBottom: Int? = null,
    val paddingStart: Int? = null,
    val paddingEnd: Int? = null,
    val fillMaxSize: Boolean? = null,
    val fillMaxWidth: Boolean? = null,
    val backgroundColor: String? = null,
    val horizontalAlignment: String? = null, // For Column: "start", "center", "end"
    val verticalAlignment: String? = null, // For Row: "top", "center", "bottom"
    val contentAlignment: String? = null, // For Box: "topStart", "center", "bottomEnd", etc.
    val verticalScroll: Boolean? = null // Enable vertical scrolling for Column
)

File: ssr-simple/ssr-simple-react/src/models/ModifierModel.ts
export interface ModifierModel {
  height?: number;
  width?: number;
  weight?: number; // For Row/Column children: proportional size
  padding?: number;
  paddingTop?: number;
  paddingBottom?: number;
  paddingStart?: number;
  paddingEnd?: number;
  fillMaxSize?: boolean;
  fillMaxWidth?: boolean;
  backgroundColor?: string;
  horizontalAlignment?: 'start' | 'center' | 'end';
  verticalAlignment?: 'top' | 'center' | 'bottom';
  contentAlignment?: 'topStart' | 'topCenter' | 'topEnd' | 'centerStart' | 'center' | 'centerEnd' | 'bottomStart' | 'bottomCenter' | 'bottomEnd';
  verticalScroll?: boolean; // Enable vertical scrolling for Column
}
